client/state/sites/selectors/get-site-plan.ts

```diff
@@ -36,16 +36,6 @@ export default function getSitePlan(
 	}
 
 	if ( site.plan?.expired ) {
-		if ( site.jetpack ) {
-			return {
-				product_id: 2002,
-				product_slug: 'jetpack_free',
-				product_name_short: 'Free',
-				free_trial: false,
-				expired: false,
-			};
-		}
```

Comment:
I don't think we can remove this. `getSitePlan` can be used for self-hosted Jetpack sites, and those with expired plans need to still report their plan as `jetpack_free`. 

Maybe we could change the check to: 

```js
if ( site.jetpack && ! site.is_wpcom_atomic ) {
  return {
    product_id: 2002,
    product_slug: 'jetpack_free',
    product_name_short: 'Free',
    free_trial: false,
    expired: false,
  };
}
```

Reply:
Good call. I've reverted the changes and added some tests to that we have covered both cases.